packages/eas-cli/src/credentials/ios/appstore/ensureAppExists.ts

```diff
@@ -83,18 +84,30 @@ export async function syncCapabilities(
   const spinner = ora(`Syncing capabilities`).start();
 
   try {
-    const capability = await bundleId.hasCapabilityAsync(CapabilityType.PUSH_NOTIFICATIONS);
-    if (!capability && options.enablePushNotifications) {
+    const notifications = await bundleId.hasCapabilityAsync(CapabilityType.PUSH_NOTIFICATIONS);
+    if (!notifications && options.enablePushNotifications) {
       await bundleId.updateBundleIdCapabilityAsync({
         capabilityType: CapabilityType.PUSH_NOTIFICATIONS,
         option: CapabilityTypeOption.ON,
       });
-    } else if (capability && !options.enablePushNotifications) {
+    } else if (notifications && !options.enablePushNotifications) {
```

Comment:
maybe in case of undefined in options you should not modify anything(both here and for associated domains)